tests/integration/s2n_handshake_test_gnutls.py

```diff
@@ -92,21 +107,24 @@ def main(argv):
                 continue
 
             # Add the SSL version to make the cipher priority string fully qualified
-            complete_priority_str = cipher_priority_str + ":+" + S2N_PROTO_VERS_TO_GNUTLS[ssl_version]
-            ret = try_gnutls_handshake(argv[0], int(argv[1]), complete_priority_str)
-            print("Cipher: %-30s Vers: %-10s ... " % (cipher_name, S2N_PROTO_VERS_TO_STR[ssl_version]), end='')
-            if ret == 0:
-                if sys.stdout.isatty():
-                    print("\033[32;1mPASSED\033[0m")
-                else:
-                    print("PASSED")
-            else:
-                if sys.stdout.isatty():
-                    print("\033[31;1mFAILED\033[0m")
-                else:
-                    print("FAILED")
-                failed = 1
-    return failed
+            complete_priority_str = cipher_priority_str + ":+" + S2N_PROTO_VERS_TO_GNUTLS[ssl_version] + ":+SIGN-ALL"
+
+            if handshake(argv[0], int(argv[1]), cipher_name, ssl_version, complete_priority_str) < 0:
+                return -1
+
+    # Produce permutations of every accepted signature alrgorithm in every possible order
+    signatures = ["SIGN-RSA-SHA1", "SIGN-RSA-SHA224", "SIGN-RSA-SHA256", "SIGN-RSA-SHA384", "SIGN-RSA-SHA512"];
+    for size in range(1, len(signatures) + 1):
+        print("\n\tTesting ciphers using signature preferences of size: " + str(size))
+
+        for permutation in itertools.permutations(signatures, size):
+
+            # Try an ECDHE cipher suite and a DHE one
+            for cipher in filter(lambda x: x.openssl_name == "ECDHE-RSA-AES128-GCM-SHA256" or x.openssl_name == "DHE-RSA-AES128-GCM-SHA256", S2N_CIPHERS):
+
+                complete_priority_str = cipher.gnutls_priority_str + ":+VERS-TLS1.2:+" + ":+".join(permutation)
```

Comment:
I think printing the current signature permutation would be useful for debugging test failures